code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -323,49 +340,50 @@
 
 
 /mob/living/carbon/human/ex_act(severity, ex_target)
-	var/b_loss = null
-	var/f_loss = null
-	var/bomb_armor = getarmor(null, "bomb")
+
 	if(istype(ex_target, /datum/spacevine_mutation) && isvineimmune(src))
 		return
+	var/b_loss = 0
+	var/f_loss = 0
+	var/bomb_armor = getarmor(null, "bomb")
 
 	switch (severity)
 		if (1)
-			b_loss += 500
-			if (prob(bomb_armor))
-				shred_clothing(1,150)
+			if(prob(bomb_armor))
+				b_loss = 500
 				var/atom/target = get_edge_target_turf(src, get_dir(src, get_step_away(src, src)))
 				throw_at(target, 200, 4)
 			else
 				gib()
 				return
 
 		if (2)
-			b_loss += 60
-
-			f_loss += 60
-			if (prob(bomb_armor))
-				b_loss = b_loss/1.5
-				f_loss = f_loss/1.5
-				shred_clothing(1,25)
-			else
-				shred_clothing(1,50)
+			b_loss = 60
+			f_loss = 60
+			if(bomb_armor)
+				b_loss = 30*(2 - round(bomb_armor*0.01, 0.05))
+				f_loss = b_loss
 
 			if (!istype(ears, /obj/item/clothing/ears/earmuffs))
 				adjustEarDamage(30, 120)
 			if (prob(70))
 				Paralyse(10)
 
 		if(3)
-			b_loss += 30
-			if (prob(bomb_armor))
-				b_loss = b_loss/2
+			b_loss = 30
+			if(bomb_armor)
+				b_loss = 15*(2 - round(bomb_armor*0.01, 0.05))
+
 			if (!istype(ears, /obj/item/clothing/ears/earmuffs))
 				adjustEarDamage(15,60)
 			if (prob(50))
 				Paralyse(8)
 
 	take_overall_damage(b_loss,f_loss)
+	if(bomb_armor == 100) //full bomb armor set, we don't call contents_explosion
+		flash_act()
+		return

```

Comment:
This is to avoid contents_explosion() calling ex_act on our clothes. I guess it's a bit of a nerf, since now wearing some bomb armor but not a full set still expose all your clothes to be shredded.